qubes-gui-client/src/buffer.rs

```diff
@@ -423,36 +426,34 @@ mod tests {
         let mut under_test = RawMessageStream::<MockVchan> {
             vchan: mock_vchan,
             queue: Default::default(),
-            offset: 0,
             state: ReadState::ReadingHeader,
             buffer: vec![],
             did_reconnect: false,
             xconf: Default::default(),
             domid: 0,
         };
         under_test.write(b"test1").unwrap();
-        assert_eq!(under_test.queue.len(), 1, "message queued");
-        assert_eq!(under_test.queue[0], b"test1");
-        assert_eq!(under_test.offset, 0, "no bytes written");
+        assert_eq!(under_test.queue.len(), 5, "message queued");
+        assert_eq!(under_test.queue.as_slices().0[0..5], *b"test1");
```

Comment:
You can compare a `VecDeque` with a slice directly, without relying on which of the two slices contains the data.
```suggestion
        assert_eq!(under_test.queue, *b"test1");
```